src/nprintml/label/aggregator/base.py

```diff
@@ -15,27 +15,34 @@ class LabelAggregator(abc.ABC):
     Inheritors of this class are expected to implement the abstract
     method `__call__()` to provide their method of feature aggregation.
 
+    Inheritors *may* extend `__init__()` so as to set instance attribute
+    `labels` to the eagerly-loaded labels, for sharing with other
+    components -- for example to avoid preparing more data than there
+    are labels.
+
     """
-    def __init__(self, npt_csv, label_csv):
-        self.npt_csv = npt_csv
+    def __init__(self, label_csv):
         self.label_csv = label_csv
+        self.labels = None
```

Comment:
The aggregators are now constructed eagerly from the label file path – and, optionally, they set the attribute `labels` to something of interest to the pipeline, namely the loaded labels.

The nPrint file path(s) must now be passed on execution, (when we have them, inside the pipeline).